18F/identity-idp
# LG 8441 Prioritize ready enrollments

## 🎫 [Ticket](https://cm-jira.usa.gov/browse/LG-8441)

Note: this ticket follows work @NavaTim completed in [LG-8440](https://cm-jira.usa.gov/browse/LG-8440). [Pull Request here](https://github.com/18F/identity-idp/pull/8403).

## 🛠 Summary of changes

1. Created two new sub-classes of GetUspsProofingResultsJob
2. Each subclass is set to be called at different frequencies via cron
3. Each subclass processes different enrollments: enrollments that have been marked 'ready' by our SQS job, and those not marked 'ready'.
4. Tests have been created for each subclass
5. Refactored GetUspsProofingResultsJob
6. Pulled config checks into separate methods
7. Turned percent error calculation into own method
8. Moved email related methods into new module
9. Moved analytics related methods into new module
10. Created wrapper methods for each analytics method
11. Removed an error handler method with duplicate internals to other error handler method

## 📜 Testing Plan

To test locally:
- Set the following config values to `true` in your local dev `application.yml` file
- [ ] in_person_proofing_enabled
- [ ] in_person_enrollments_ready_job_enabled
- Check that the `GetUspsReadyProofingResultsJob` and `GetUspsWaitingProofingResultsJob` are performed in the `workers.log` file located at `log/workers.log`
